testing/color.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2023 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package testing
+
+import (
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+)
+
+func init() {
+	if _, ok := os.LookupEnv("COLOR_DIFF"); ok {
```

Comment:
**[issue, suggestion]:** I get a coloured diff even if `COLOR_DIFF` is not set. It seems that `COLOR_DIFF` does not override faith/color's https://github.com/fatih/color#disableenable-color mechanics. Since faith/color is already being clever depending on the connected output, couldn't we drop `COLOR_DIFF` and let the fatih/color handle it? After all, `NO_COLOR` would be a way to disable it.

Reply:
interesting, I while testing I could never get the color to be active by default. I assumed there was something about `go test` is executing that triggered the default to no color. I added COLOR_DIFF to force color on.

How are you executing the code? Version of go, os, console.

Reply:
```
❯ go version
go version go1.21.0 darwin/amd64

❯ uname -a
Darwin ***** 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; root:xnu-8796.141.3~6/RELEASE_X86_64 x86_64

❯ # terminal
❯ kitty --version
kitty 0.29.2 created by Kovid Goyal

❯ # run tests
❯ go run github.com/onsi/ginkgo/v2/ginkgo --keep-going -r
...
```

Reply:
I see colorized output by default with both kitty and ginkgo, but using either a different test runner or a different terminal and it's all plain text.

`NO_COLOR` is only used to disable color output. `NO_COLOR=true` and `NO_COLOR=false` are equivalent ways to disable color output. I'm using `COLOR_DIFF` as a way to override the tty detection and force colorized output. I'm up for ways to be smarter about the default detection.

Reply:
@scothis Thanks for running the experiments and making the call. Yay colors!